src/services/rateSource/PrivatBankService.ts

```diff
@@ -0,0 +1,37 @@
+import { RateSourceResponse, RateSourceService } from './RateSourceService';
+import responseMessages from '../../constants/responseMessages';
+
+type PrivatBankResponse = {
+  ccy: 'EUR' | 'USD';
+  base_ccy: 'UAH';
+  buy: number;
+  sale: number;
+}[];
+
+const { SOMETHING_WENT_WRONG } = responseMessages;
+
+class PrivatBankService extends RateSourceService {
+  constructor() {
+    super('https://api.privatbank.ua/p24api/pubinfo');
+  }
+
+  async retrieve(): Promise<RateSourceResponse> {
+    const rawResponse = await fetch(this.url);
+
+    const [, USDRate] = (await rawResponse?.json()) as PrivatBankResponse;
```

Comment:
Are these parenthesis around await rawResponse?.json() needed?

Same comment to other services

Reply:
That's my prettier rule 😌